activerecord/test/cases/core_test.rb

```diff
@@ -51,9 +65,9 @@ def test_inspect_class_without_table
     assert_equal "NonExistentTable(Table doesn't exist)", NonExistentTable.inspect
   end
 
-  def test_inspect_relation_with_virtual_field
-    relation = Topic.limit(1).select("1 as virtual_field")
-    assert_match(/virtual_field: 1/, relation.inspect)
```

Comment:
I decided to remove this test. `Relation#inspect` calls `inspect` on each of the records. It seems unnatural that virtual columns would be included in `attributes_to_expect` so I didn't want to add that as a test case.

I think we could add a `full_inspect` method to `Relation` if we want to preserve this behaviour.

Reply:
I think you can rewrite this test in a way that makes sure virtual attributes are inside the loaded record and can be inspected. You could call `full_inspect` in `relation.first`